### PR TITLE
Enhancement/name gk type

### LINKED ISSUE #51: Minor : Pyro init takes gk_type but then we set gk_code to change type
Fairly minor but I always forget that the init call takes gk_type but we change the type of the instance by setting the gk_code property. Could these have the same name?

### PR DESCRIPTION
**Pull request by bpatel2107**
Fixes #51

### CODE REVIEW

File: pyrokinetics/examples/example.py [hunk 1]
```diff
@@ -2,14 +2,12 @@
 
 gs2input = template_dir / "input.gs2"
 eqfile = template_dir / "test.geqdsk"
-pyro = Pyro(eq_file=eqfile, eq_type="GEQDSK", local_geometry="Miller", gk_type="GS2")
+pyro = Pyro(gk_code="GS2", gk_file=gs2input)
```
Comment (ZedThree) [inline pyrokinetics/examples/example.py:5]: `eqfile` is now not used?

File: pyrokinetics/pyro.py [hunk 2]
```diff
@@ -31,8 +32,12 @@ def __init__(
         self.supported_local_geometries = ["Miller", None]
 
         self.gk_file = gk_file
-        self.gk_type = gk_type
-        self.gk_code = self.gk_type
+        if gk_type is not None and gk_code is None:
+            raise DeprecationWarning(
+                "gk_type is no longer used, please use gk_code instead"
+            )
```
Comment (ZedThree) [inline pyrokinetics/pyro.py]: Should probably also keep `self.gk_code = gk_type` for now and remove in a future version?

**Review by ZedThree — COMMENTED** [review]
Might be good to note the change in CHANGELOG.md?

File: pyrokinetics/examples/example.py (hunk 1 above)
Comment (bpatel2107) [reply pyrokinetics/examples/example.py:5]: It wasn't really being used before. A lot of the example needed updating

File: pyrokinetics/pyro.py (hunk 2 above)
Comment (bpatel2107) [reply pyrokinetics/pyro.py]: How do I "raise" the warning without stopping the program. Can't seem to figure that out

Comment (ZedThree) [reply pyrokinetics/pyro.py]: The `warnings` module is what you want:

https://docs.python.org/3/library/warnings.html#warnings.warn

This users filter the warning out

**Comment by bpatel2107** [comment]
I don't think we have a CHANGELOG.md. Can add it to the base directory

**Comment by bpatel2107** [comment]
Any extra thoughts on the changelog @ZedThree ?

**Comment by ZedThree** [comment]
LGTM :+1:

**Review by ZedThree — APPROVED** [review]

**Pull request merged**